test/test_intensify/test_successive_halving.py

```diff
@@ -1471,13 +1471,32 @@ def _exhaust_stage_execution(self, intensifier, taf, challengers):
         robust against this scenario
         """
         pending_processing = []
-        toggle = True
+        stage = 0 if not hasattr(intensifier, 'stage') else intensifier.stage
+        curr_budget = intensifier.all_budgets[stage]
+        prev_budget = int(intensifier.all_budgets[stage - 1]) if stage > 0 else 0
+        if intensifier.instance_as_budget:
+            total_runs = int(curr_budget - prev_budget) * int(
+                intensifier.n_configs_in_stage[stage])
+            toggle = np.random.choice([True, False], total_runs).tolist()
+            while not np.any(toggle) or not np.any(np.invert(toggle)):
+                # make sure we have both true and false!
```

Comment:
But this doesn't ensure having at least one `True` and one `False`, it just increases your chances. Could you manually set one to True or False if necessary?